### ISSUE TITLE
Read response headers

### ISSUE THREAD

**Issue by hdgarrood** [slamdata/purescript-affjax#30]
I'd like to be able to do things like ask if a response has a certain header, or get the value of a certain header in the response (if it exists).

Currently it seems impossible to do this, since the `ResponseHeader` data type doesn't export its constructor, and I don't see any functions for extracting this information from them.

**Comment by hdgarrood** [comment]
Oh - having seen `RequestHeader`, I suppose the motivation is to avoid a stringly-typed interface?

Would it be possible to expose some way of getting everything as just strings, possibly with a note in the documentation suggesting people use the stronger typed interface instead?

**Comment by garyb** [comment]
Yeah, exactly. Both types need work actually to be extended with common cases, but we could definitely do something to expose them as strings too - `responseHeaderName :: ResponseHeader -> String`, `responseHeaderValue :: ResponseHeader -> String` like there is for `requestHeaderName`. Would that do it?

**Comment by hdgarrood** [comment]
Yeah I think so - then presumably I could do something like `response.headers # Data.Foldable.find (responseHeaderName >>> eq "Set-Cookie") # map responseHeaderValue` to get the header value as a Just if it's there, and Nothing otherwise?

Also perhaps a newtype wrapper around `String` which makes equality and comparison case insensitive might be helpful here?

**Comment by garyb** [comment]
Yep, that's along the lines of what I was thinking. Case insensitive newtype for `String` comparisons sounds good too, would make a nice addition to `purescript-strings`.

**Comment by paf31** [comment]
Cf. http://hackage.haskell.org/package/case-insensitive :+1:

**Comment by garyb** [comment]
Ok, I've added those functions now. Although on further reflection neither `responseHeaderValue` or `requestHeaderValue` are quite right in the long run, there should probably be a phantom type on `ResponseHeader` and `RequestHeader` that determines their contained content type.
